std/experimental/ndslice/selection.d

```diff
@@ -98,6 +98,23 @@ template pack(K...)
 ///
 @safe @nogc pure nothrow unittest
 {
+    import std.experimental.ndslice : sliced, Slice, pack;
+    import std.range : iota;
+
+    auto r = (3 * 4 * 5 * 6).iota;
+    auto a = r.sliced(3, 4, 5, 6);
+    auto b = a.pack!2;
+    
+    assert(b.shape == [3, 4]);
```

Comment:
this may cause GC allocation, you can circumvent it by e.g. using `enum res = [3, 4]`